src/llfs/storage_file_builder.test.cpp

```diff
@@ -56,6 +58,123 @@ class StorageFileBuilderTest : public ::testing::Test
+        ASSERT_EQ(page_device_config.page_0_offset, expected_page0_offset);
+        ASSERT_EQ(page_device_config.device_id, expected_device_id);
+        ASSERT_EQ(page_device_config.page_count, (little_i64)kTestPageCount);
+        ASSERT_EQ(page_device_config.page_size_log2, page_size_log2);
+
+        // Update page offset numbers and device ID
+        //
+        expected_page0_offset +=
+            page_device_config.page_size() * kTestPageCount - llfs::PackedConfigSlot::kSize;
+        expected_device_id += 1;
+      }
+
+      // Assuming 1 slot filled if back slot, all slots filled otherwise.
+      //
+      u64 expected_num_slots =
+          (config_block == config_blocks.back()) ? 1 : llfs::PackedConfigBlock::kMaxSlots;
+
+      // Set offsets of config block based on position
+      //
+      if (config_block == config_blocks.front()) {
+        expected_prev_offset = llfs::PackedConfigBlock::kNullFileOffset;
+      } else {
+        expected_prev_offset = -expected_next_offset;
+      }
+      if (config_block == config_blocks.back()) {
+        expected_next_offset = llfs::PackedConfigBlock::kNullFileOffset;
+      } else {
+        expected_next_offset = batt::round_up_bits(
+            12, expected_page0_offset + llfs::PackedPageDeviceConfig::kSize * expected_num_slots);
+      }
+
+      // Verify the packed config block
+      //
+      ASSERT_EQ(p_config_block.magic, llfs::PackedConfigBlock::kMagic);
+      ASSERT_EQ(p_config_block.version, llfs::make_version_u64(0, 1, 0));
+      ASSERT_EQ(p_config_block.prev_offset, expected_prev_offset);
+      ASSERT_EQ(p_config_block.next_offset, expected_next_offset);
+      ASSERT_EQ(p_config_block.slots.size(), expected_num_slots);
+      ASSERT_EQ(p_config_block.crc64, p_config_block.true_crc64());
+    }
+  }
+
+  // Writes random data to disk given the file, offset, and amount of data to write. write_offset
+  // and write_size must be multiples of 512.
+  //
+  llfs::Status write_rand_data(llfs::IoRingRawBlockFile& test_file, i64 write_offset,
+                               i64 write_size)
+  {
+    constexpr i64 kMaxBufferSize = 32768;
+    constexpr i64 kBufferAlign = 512;
+    if (write_offset % kBufferAlign != 0 || write_size % kBufferAlign != 0) {
+      return batt::Status{batt::StatusCode::kInvalidArgument};
+    }
+
+    std::aligned_storage_t<kMaxBufferSize, kBufferAlign> rand_data;
+
+    std::random_device rd;
+    std::mt19937_64 gen(rd());
+    std::uniform_int_distribution<u64> distrib(std::numeric_limits<u64>::min(),
+                                               std::numeric_limits<u64>::max());
+
+    while (write_size > 0) {
+      u64 buffer_size = std::min(kMaxBufferSize, write_size);
+
+      llfs::MutableBuffer read_buffer(&rand_data, buffer_size);
+
+      for (u64 i = 0; i < (buffer_size / sizeof(u64)); i++) {
+        u64 temp = distrib(gen);
+        std::memcpy(read_buffer.data(), &temp, sizeof(temp));
```

Comment:
The union variable requires that size of the array is defined by a constant, so I went back and made it the size of "kMaxBufferSize / sizeof(u64)" as you suggested. This means that even if only 512 more bytes are left to write, it will still populate the rand_data with kMaxBufferSize bytes. Is this ok?

Reply:
Hmm, good point.  Well, I suppose that means the loop I gave above probably is too simple; we should not always iterate over all elements of `rand_data.items`, rather only enough to generate the needed amount of pseudo-random data.  I think that's the best way to go.